spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -127,6 +129,16 @@ class SparkWrite {
     this.dsSchema = dsSchema;
     this.extraSnapshotMetadata = writeConf.extraSnapshotMetadata();
     this.partitionedFanoutEnabled = writeConf.fanoutWriterEnabled();
+
+    if (writeConf.outputSpecId() == null) {
```

Comment:
Why not have this inside `SparkWriteConf` and make `outputSpecId()` return `int`?

Reply:
We could keep a reference to `Table`, just like we do in `SparkReadConf`.